src/models/room.js

```diff
@@ -201,6 +204,27 @@ Room.prototype.getLiveTimeline = function() {
     return this.getUnfilteredTimelineSet().getLiveTimeline();
 };
 
+/**
+ * Get the lazy loading state, whether loading is needed or not.
+ * @return {bool} whether or not the members of this room need to be loaded
+ */
+Room.prototype.membersNeedLoading = function() {
+    return this._membersNeedLoading;
+};
+
+/**
+ * Sets the lazily loaded members from the result of calling /joined_members
+ * @param {Promise} joinedMembersPromise promise with result of /joined_members endpoint
+ */
+Room.prototype.setLazilyLoadedMembers = async function(joinedMembersPromise) {
+    this._membersNeedLoading = false;
+    const members = await joinedMembersPromise;
+    //wait 10 seconds
+    await new Promise((resolve) => setTimeout(resolve, 5000));
```

Comment:
Why should this wait? `await joinedMembersPromise` should make sure that it only continues executing once it is (successfully) done.
Besides that "5000" != 10 seconds ;)